tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/OpenTelemetry/StartRootSpanIntegration.cs

```diff
@@ -0,0 +1,77 @@
+        TypeName = "OpenTelemetry.Trace.Tracer",
+        MethodName = "StartRootSpan",
+        ReturnTypeName = "OpenTelemetry.Trace.TelemetrySpan",
+        ParameterTypeNames = new[] { ClrNames.String, "OpenTelemetry.Trace.SpanKind", "OpenTelemetry.Trace.SpanAttributes", "System.Collections.Generic.IEnumerable`1[OpenTelemetry.Trace.Link]", "System.DateTimeOffset" },
+        MinimumVersion = "1.0.0",
+        MaximumVersion = "1.0.0",
+        IntegrationName = nameof(Configuration.IntegrationId.OpenTelemetry))]
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class StartRootSpanIntegration
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(StartSpanIntegration));
+
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTracer">Type of the tracer</typeparam>
+        /// <typeparam name="TSpanKind">Type of the span kind</typeparam>
+        /// <typeparam name="TSpanAttributes">Type of the span attributes</typeparam>
+        /// <typeparam name="TLinks">Type of the span links</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="name">Span name.</param>
+        /// <param name="kind">Span kind.</param>
+        /// <param name="spanAttributes">Initial attributes for the span.</param>
+        /// <param name="links">Links for the span.</param>
+        /// <param name="startTimeinstance">Start time for the span.</param>
+        /// <returns>Calltarget state value</returns>
+        internal static CallTargetState OnMethodBegin<TTracer, TSpanKind, TSpanAttributes, TLinks>(TTracer instance, string name, TSpanKind kind, TSpanAttributes spanAttributes, TLinks links, DateTimeOffset startTimeinstance)
+        {
+            Log.Information("Entered OnMethodBegin for StartRootSpan");
+            return new CallTargetState(Tracer.Instance.InternalActiveScope);
+        }
+
+        /// <summary>
+        /// OnMethodEnd callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TReturn">Type of the return value</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="returnValue">Return value.</param>
+        /// <param name="exception">Exception instance in case the original code threw an exception.</param>
+        /// <param name="state">Calltarget state value</param>
+        /// <returns>CallTargetReturn</returns>
+        internal static CallTargetReturn<TReturn> OnMethodEnd<TTarget, TReturn>(TTarget instance, TReturn returnValue, Exception exception, in CallTargetState state)
+        {
+            Log.Information("Entered OnMethodEnd for StartRootSpan");
+
+            // If the integration created a new scope, dispose it
```

Comment:
Should we be checking that the activity integration is enabled first?

Reply:
Added in 5e6024b64f5cad7671d2a9256f0b60ebe69ea95c